include/circt/Dialect/FIRRTL/FIRRTLDeclarations.td

```diff
@@ -161,6 +161,57 @@ def InstanceOp : HardwareDeclOp<"instance", [
   }];
 }
 
+def InstanceChoiceOp : HardwareDeclOp<"instance_choice", [
+    DeclareOpInterfaceMethods<SymbolUserOpInterface>
+  ]> {
+  let summary = "Creates an instance of a module based on a option";
+
+  let description = [{
+    The instance choice operation creates an instance choosing the target based
+    on the value of an option if one is specified, instantiating a default
+    target otherwise.
+
+    The port lists of all instance targets must match exactly.
+
+    Examples:
+    ```mlir
+    %0 = firrtl.instance_choice foo @Foo,
+      @FpgaFoo if @Opt::@FPGA
+      (in io: !firrtl.uint)
```

Comment:
Is this assembly format correct? Does this not get an "or" if there is only one option? (If it is different for one vs. multiple options, then an example showing both would be beneficial. I have some nits about the use of "or" later and would prefer something that's just dictionary-like.)